src/components/Secrets/SecretsForm/ApplicationDropdown.tsx

```diff
@@ -18,7 +18,12 @@ export const ApplicationDropdown: React.FC<React.PropsWithChildren<ApplicationDr
 
   const dropdownItems = React.useMemo(
     () =>
-      loaded ? applications.map((a) => ({ key: a.metadata.name, value: a.metadata.name })) : [],
+      loaded
+        ? applications.map((a) => ({
+            key: a.metadata.name,
+            value: a.spec?.displayName ?? a.metadata.name,
```

Comment:
nit: maybe we could use the `getApplicationDisplayName` util function here (`value: getApplicationDisplayName (a)`) :)

Reply:
updated